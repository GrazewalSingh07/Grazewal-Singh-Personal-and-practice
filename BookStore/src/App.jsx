 
 
import {Routes,Route} from "react-router-dom"
import { Navbar } from './components/Navbar'
import { Books } from './Pages/Books'
import { SingleBook } from './Pages/SingleBook'
import { EditBook } from './Pages/EditBook'
import { Login } from './Pages/Login'
import {PrivateRoute} from "./components/PrivateRoute"
function App() {
 

  return (
    <div className="App">
      <Navbar/>
      <Routes>

        <Route path='/' element={<PrivateRoute><Books/></PrivateRoute>}></Route>
        <Route path='/Book/:id' element={<PrivateRoute><SingleBook/></PrivateRoute>}></Route>
        <Route path='/Book/:id/edit' element={<PrivateRoute><EditBook/></PrivateRoute>}></Route>
        <Route path='/login' element={<Login/>}></Route>
        <Route path="*" element={<h3>Page Not Found</h3>} />
      </Routes>
       
    </div>
  )
}

export default App
