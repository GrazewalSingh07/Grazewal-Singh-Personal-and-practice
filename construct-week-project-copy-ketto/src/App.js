import Header from "./Components/Header";
import MainRoutes from "./Pages/MainRoutes";

function App() {
  return (
    <div>
      <Header/>
      <MainRoutes />
    </div>
  );
}

export default App;
