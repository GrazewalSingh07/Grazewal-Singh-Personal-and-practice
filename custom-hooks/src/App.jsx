import { useState } from "react"
import { useFetch } from "./hooks/useFetch"
import { useSearch } from "./hooks/useSearch"
import { useTimeout } from "./hooks/useTimeout"

 

function App() {
  const {loading,data,error}=useFetch("https://fakestoreapi.com/products")
 const {ready}=useTimeout(5000)

 const [input,setinput]=useState("")
 const {users}=useSearch(input,"https://api.github.com/search/users")
 console.log(users)
  return (
    <div className="App">
      <input type="text" placeholder="search user" onChange={(e)=>setinput(e.target.value)}/>
      {ready&&"Ready"}
      {!ready&&"Not ready"}
    </div>
  )
}

export default App
