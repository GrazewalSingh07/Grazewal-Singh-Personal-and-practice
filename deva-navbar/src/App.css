 .App{
  margin: auto;
  padding: 2rem;
  background-color: aqua;
  width: 90%;
  display: flex;
  justify-content: space-around;
 }