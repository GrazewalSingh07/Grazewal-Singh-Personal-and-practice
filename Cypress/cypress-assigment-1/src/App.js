 
import './App.css';
import { Counter } from './component/Counter';
import { Form } from './component/Form';

function App() {
  return (
    <div className="App">
      <Counter/>
      <Form/>
    </div>
  );
}

export default App;
