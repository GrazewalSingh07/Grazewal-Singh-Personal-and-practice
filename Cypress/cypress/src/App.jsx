 
import { useState } from 'react';
import './App.css';

function App() {
  const [counter,setcounter]=useState(0)
  return (
    <div className="App">
       <h2>Count is {counter}</h2>
       <button className='increment_count' onClick={()=>setcounter((prev)=>prev+1)}>inc</button>
       <button className='decrement_count' onClick={()=>setcounter((prev)=>prev+-1)}>dec</button>
    </div>
  );
}

export default App;
