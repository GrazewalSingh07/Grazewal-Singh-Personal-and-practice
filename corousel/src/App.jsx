import { useState } from 'react'
import logo from './logo.svg'
import './App.css'
import CaptionCarousel from './Corousel'

function App() {
 

  return (
    <div className="App">
       <CaptionCarousel/>
    </div>
  )
}

export default App
