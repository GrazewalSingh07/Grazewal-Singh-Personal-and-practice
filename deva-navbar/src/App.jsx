import { Button } from "./Components/Button"
import { Links } from "./Components/Links"
import { Logo } from "./Components/Logo"

 import "./App.css"
function App() {
  

  return (
    <div className="App">
       <Logo width="50px"/>
       <Links/>
       <Button>Sign in</Button>
    </div>
  )
}

export default App
