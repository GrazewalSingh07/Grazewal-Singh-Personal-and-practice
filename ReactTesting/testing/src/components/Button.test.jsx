import {render,screen,fireEvent} from "@testing-library/react"
import App from "../App"
import { Button } from "./Button"
import renderer from "react-test-renderer"
import userEvent from "@testing-library/user-event"
//this render will test our dom virtually. it is like fake rendered method, it will not create dom in actual
describe("Button Test",()=>{
    it("Should have button dom",()=>{
        render(<Button>Click Me</Button>);
        // screen.debug()

        let button=screen.getByText("Click Me") //screen.getbyttest is like document.getelementbyid
        expect(button).toBeInTheDocument()
        // second try
       
    })
    it(" should have button with text Click Me in app.js",()=>{
        render (<App></App>)
        let button=screen.getByTestId("cButton")
        expect(button).toBeInTheDocument()
        // in above test
        //if i endered app and in app we have other element with text "Click Me" and button didn't had "Click Me" but the test still passed
        // so  using getByText is not a good idea

    })
    it("It should render empty button,",()=>{
        render(<Button></Button>)
        let button=screen.getByTestId("cButton");
        expect(button).toBeEmptyDOMElement()
        //it will check if Button has any children
    })
    it("It should change theme",()=>{
        render(<App></App>)
        let h3= screen.getByText("Current Theme is light")
        expect(h3).toHaveTextContent("light")

        let themeButton=screen.getByText("Change Theme");
        userEvent.click(themeButton)// userEvent mimic the behaviour of user click a button
        // we can use fireEvent in place of userEvent as well
        // It is same thing
        expect(h3).toHaveTextContent("dark")
    })
    it("Should call the given function",()=>{
        const mockfn=jest.fn()
        // we dont care what this function is  
        render(<Button onclick={mockfn}> Click Me</Button>)
        const btn=screen.getByTestId("cButton")
        fireEvent.click(btn) //or we can write userEvent.click(btn)
        // expect(mockfn).toBeCalled()
        fireEvent.click(btn)
        expect(mockfn).toBeCalledTimes(2)// it check how many times a particular btn is clicked
    })

    //SnapShot -->
    // Screenshots are Represented  in a particular formate
    it("should match snapshot",()=>{
       const tree= renderer.create(<Button colorScheme={"green"} variant={"ghost"}>Add counter</Button>).toTree()
       expect(tree).toMatchSnapshot()
    })
    /// if someone chaged anything in btn component
    // it will fail
    // and ask if we want to overwrite
   // we press w for options
   // and them press u to update snapshot
// we can use toJSON or toTree 
// toTree gives more details
})
