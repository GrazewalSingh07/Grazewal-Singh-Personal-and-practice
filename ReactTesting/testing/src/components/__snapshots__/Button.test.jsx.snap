// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Test should match snapshot 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "children": "Add counter",
    "colorScheme": "green",
    "variant": "ghost",
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "children": "Add counter",
      "className": "btn green ghost",
      "data-testid": "cButton",
      "onClick": undefined,
    },
    "rendered": Array [
      "Add counter",
    ],
    "type": "button",
  },
  "type": [Function],
}
`;
